### PR TITLE
Do not return before write stream fully closes

### PR DESCRIPTION
**Pull request by pauliusuza**
Related to #106

### CODE REVIEW

File: lib/filesystem.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ class Filesystem {
         const stream = fs.createReadStream(p)
 
         stream.pipe(tr).pipe(out)
-        return tr.on('end', function () {
```
Comment (develar) [inline lib/filesystem.js]: `finish` should be used.

**Comment by develar** [comment]
@pauliusuza this issue is not related to electron-builder issue because transform is not used by electron-builder.

File: lib/filesystem.js (hunk 1 above)
Comment (develar) [inline lib/filesystem.js]: I am wrong, "close" is correct, https://github.com/jprichardson/node-fs-extra/issues/326

**Comment by fantasydr** [comment]
I tried to use custom transform and got unstable result. It seems the stream is not closed properly.
This PR fixed the issue.
Code to repro:
```
var asar = require('asar');
var Transform = require('stream').Transform;
var fs = require('fs');

class CustomTransform extends Transform {
    _transform(chunk, enc, cb) {
        console.log(chunk.length, enc, cb); // do nothing, just print logs
        this.push(chunk);
        cb();
    };
}

function transform(filename) {
    console.log(filename);
    return new CustomTransform();
}

var src = 'input/';
var dest = 'app.asar';
asar.createPackageWithOptions(
    src, dest, 
    { transform }, 
    function() { 
        console.log('done.');
    }
)
```

**Comment by njbmartin** [comment]
Surprised this hasn't been merged yet. This fixes the issues we've had transforming multiple files.

**Comment by benlancaster** [comment]
👍  This bug is driving me nuts

**Comment by pauliusuza** [comment]
Thanks for merging 👍

**Pull request merged**